includes/data-events/connectors/ga4/class-ga4.php

```diff
@@ -21,13 +22,25 @@
  */
 class GA4 {
 
+	/**
+	 * The events being watched.
+	 *
+	 * @var array
+	 */
+	public static $watched_events = [
+		'reader_logged_in',
+		'reader_registered',
+	];
+
 	/**
 	 * Initialize the class and registers the handlers
 	 *
 	 * @return void
 	 */
 	public static function init() {
-		Data_Events::register_handler( [ __CLASS__, 'handle_reader_logged_in' ], 'reader_logged_in' );
+		foreach ( self::$watched_events as $event_name ) {
+			Data_Events::register_handler( [ __CLASS__, 'handle_' . $event_name ], $event_name );
+		}
```

Comment:
Why not make it a global handler that watches every event? A global handler exists mostly for analytics integrations like this one.

Reply:
I'll explore it

I want to have each event handler in its own callback for the sake of organization, I didn't want to have one big callback with a switch case or anything like it. But let me give it a go